sdk/storage/storage-blob/src/ContainerClient.ts

```diff
@@ -937,11 +937,39 @@ export class ContainerClient extends StorageClient {
    */
   public getPageBlobClient(blobName: string): PageBlobClient {
     return new PageBlobClient(
-      appendToURLPath(this.url, encodeURIComponent(blobName)),
+      appendToURLPath(this.url, this.escapePath(blobName)),
       this.pipeline
     );
   }
 
+  /**
+   * Escape the blobName but keep path separator ('/'). Exactly like in the dotnet SDK client.
+   */
+  private escapePath(blobName: string): string {
+    blobName = this.trim(blobName, "/");
```

Comment:
The trim operation will change blob name like '////a' to ''a'.  The SDK should not have such behavior of modifying input blob name.

Reply:
Hmm, but it is the same as in the dotnet sdk as you can see here: https://github.com/Azure/azure-sdk-for-net/blob/0ce8e17b4dee01d7d3c2e0aff6977706d448eee0/sdk/storage/Azure.Storage.Common/src/Shared/StorageExtensions.cs#L12-L28

Reply:
I opened an issue for .Net.  For JS SDK, I'd prefer to keep current behavior instead of making a breaking.  I'll talk with .Net SDK team about the inconsistent behavior.

Reply:
@mwiedemeyer , .Net SDK has merged this PR https://github.com/Azure/azure-sdk-for-net/pull/30271/files, in the PR the BlobUriBuilder would preserve slashes.

The PR I sent to you would resolve the PR checking failures. Would you take it? After that, the PR should be able to pass checking and then we can take it.

Reply:
I cannot find any PR you send to me!? Can you please tell me which changes are needed for the checks to pass?
And then I should also implement the same `PreserveBlobNameChanges` property as in the .net sdk?

Reply:
@mwiedemeyer , could you take a look at this PR?
https://github.com/mwiedemeyer/azure-sdk-for-js/pull/1 

.Net SDK has merged a PR to preserve slashes https://github.com/Azure/azure-sdk-for-net/pull/30271/files, no more action needed.

Reply:
I've updated the PR but some checks still fail. Can you please advise what I need to do?